### DISCUSSION TITLE
SWS do not use settings from static-web-server.socket

### DISCUSSION THREAD

**Discussion by bubnenkoff** [static-web-server/static-web-server#230] [Q&A] [answered]
Is seems that SWS do not use settings from https://github.com/static-web-server/static-web-server/blob/master/systemd/static-web-server.socket

I can get my server works with:
```
/usr/local/bin/static-web-server --host=0.0.0.0 --config-file /var/www/html --root /var/www/html
```
But if running with systemd I can't open site.
Possible I should set `--host=0.0.0.0` somewhere in systemd configs, but I do not see where

adding: and restarting service
```
SERVER_HOST=0.0.0.0
```
to `etc_default_static-web-server` did not get any effect

**Comment by bubnenkoff** [comment]
Hm... I still can't understand if service is staring...
I redone all on fresh install. Latest steps:
```
[root@320099 system]# systemctl enable static-web-server.socket
Created symlink /etc/systemd/system/sockets.target.wants/static-web-server.socket → /etc/systemd/system/static-web-server.socket.
[root@320099 system]# systemctl start  static-web-server.socket
[root@320099 system]# mkdir /var/www/
[root@320099 system]# mkdir /var/www/html
[root@320099 system]# systemctl start static-web-server.service
[root@320099 system]# systemctl status static-web-server.service
● static-web-server.service - Static Web Server
     Loaded: loaded (/etc/systemd/system/static-web-server.service; disabled; vendor preset: disabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-06-25 21:26:13 MSK; 582ms ago
TriggeredBy: ● static-web-server.socket
    Process: 534 ExecStart=/usr/local/bin/static-web-server --fd 0 (code=exited, status=216/GROUP)
   Main PID: 534 (code=exited, status=216/GROUP)
        CPU: 3ms
[root@320099 system]#
```
it seems that service is not working... I can't find reason of error: ` (code=exited, status=216/GROUP)`

**Reply by joseluisq** [reply]
This is because `www-data` user used in `SupplementaryGroups` probably does not exist in your system (Arch Linux).  So If you use an existing user then it should work.
For example, you could create a `nologin` user with specific privileges for the `/var/www/html` dir.

**Comment by bubnenkoff** [comment]
I got it work. Tomorrow I will describe how

**Reply by joseluisq** [reply, marked as answer]
BTW I improved the docs by providing a service example that should help others.
https://static-web-server.net/features/file-descriptor-socket-passing/#service-example

**Comment by bubnenkoff** [comment]
I checked your example. I have nothing to add. Thanks!
